examples/navier-stokes/densitycurrent.h

```diff
@@ -222,56 +222,53 @@ static int DC(void *ctx, CeedInt Q,
-                                   (dq[2][3][i] - drho[2]*u[2]) / rho}
-                                 };
-    const CeedScalar dE[3]     =  { dq[0][4][i],
+    const CeedScalar du[3][3]   = {{(dq[0][1][i] - drho[0]*u[0]) / rho,
+                                    (dq[1][1][i] - drho[1]*u[0]) / rho,
+                                    (dq[2][1][i] - drho[2]*u[0]) / rho},
+                                   {(dq[0][2][i] - drho[0]*u[1]) / rho,
+                                    (dq[1][2][i] - drho[1]*u[1]) / rho,
+                                    (dq[2][2][i] - drho[2]*u[1]) / rho},
+                                   {(dq[0][3][i] - drho[0]*u[2]) / rho,
+                                    (dq[1][3][i] - drho[1]*u[2]) / rho,
+                                    (dq[2][3][i] - drho[2]*u[2]) / rho}
+                                  };
+    const CeedScalar dE[3]      = { dq[0][4][i],
                                     dq[1][4][i],
                                     dq[2][4][i]
                                   };
     // -- Interp-to-Interp qdata
-    const CeedScalar wJ        =    qdata[0][i];
+    const CeedScalar wJ         =   qdata[0][i];
     // -- Interp-to-Grad qdata
-    //      Symmetric 3x3 matrix
-    const CeedScalar wBJ[3][3] = {{qdata[1][i],
-                                   qdata[2][i],
-                                   qdata[3][i]},
-                                  {qdata[4][i],
-                                   qdata[5][i],
-                                   qdata[6][i]},
-                                  {qdata[7][i],
-                                   qdata[8][i],
-                                   qdata[9][i]}
-                                 };
-    // -- Grad-to-Grad qdata
-    const CeedScalar wBBJ[3][3] = {{qdata[10][i],
-                                    qdata[11][i],
-                                    qdata[12][i]},
-                                   {qdata[11][i],
-                                    qdata[13][i],
-                                    qdata[14][i]},
-                                   {qdata[12][i],
-                                    qdata[14][i],
-                                    qdata[15][i]}
+    // ---- Inverse of change of coordinate matrix: X_i,j
+    const CeedScalar dXdx[3][3] = {{qdata[1][i],
+                                    qdata[2][i],
+                                    qdata[3][i]},
+                                   {qdata[4][i],
+                                    qdata[5][i],
+                                    qdata[6][i]},
+                                   {qdata[7][i],
+                                    qdata[8][i],
+                                    qdata[9][i]}
                                   };
+    // -- Grad-to-Grad qdata
+    // ---- dXdx_j,k * dXdx_k,j
+    CeedScalar dXdxdXdxT[3][3];
+    for (int j=0; j<3; j++) {
+      for (int k=0; k<3; k++) {
+        dXdxdXdxT[j][k] = dXdx[j][k] * dXdx[k][j];
```

Comment:
It's supposed to be matrix multiplication, so perhaps `dXdx[j][l] * dXdx[k][l]` (summed over `l`).

Reply:
Of course! Sorry about that!